### PR TITLE
feat: APP-3053 - Implement Tabs core component

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description

Implement Tabs core component + stories & tests

Task: [APP-3053](https://aragonassociation.atlassian.net/browse/APP-3053)

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.

[APP-3053]: https://aragonassociation.atlassian.net/browse/APP-3053?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        'group flex cursor-pointer items-center gap-x-4 border-primary-400 py-3 text-base font-normal leading-tight text-neutral-500', // base
+        'hover:text-neutral-600', // hover
+        'active:data-[state=active]:text-neutral-800 active:data-[state=active]:shadow-[inset_0_0_0_0,0_1px_0_0] active:data-[state=active]:shadow-primary-400', // active click
+        'focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset', // focus
+        { 'hover:shadow-[inset_0_0_0_0,0_1px_0_0] hover:shadow-neutral-800': isUnderlined }, //  isUnderlined variant
+        'data-[state=active]:text-neutral-800 data-[state=active]:shadow-[inset_0_-1px_0_0,0_1px_0_0] data-[state=active]:shadow-primary-400', // active selection
+        className,
+    );
+
+    const iconClassNames = classNames(
```
Comment (Fabricevladimir) [inline src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:35]: Icon is neutral-500 when selected.

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { TabsTrigger as RadixTabsTrigger, type TabsTriggerProps as RadixTabsTriggerProps } from '@radix-ui/react-tabs';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { Icon, type IconType } from '../../icon';
+import { TabsContext } from '../tabsRoot/tabsRoot';
+export interface ITabsTriggerProps extends RadixTabsTriggerProps {
+    /**
+     * The label of the tab.
+     */
+    label: string;
```
Comment (Fabricevladimir) [inline src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:10]: Is there a case for a tab without a label?

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Value of the tab is meant to pair with the Tabs.Content value prop.
+     */
+    value: string;
```
Comment (Fabricevladimir) [inline src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:14]: Should value be required if tabs can be used without content?

File: src/core/components/tabs/tabsList/tabsList.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { TabsList as RadixTabsList, type TabsListProps as RadixTabsListProps } from '@radix-ui/react-tabs';
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { TabsContext } from '../tabsRoot/tabsRoot';
+
+export interface ITabsListProps extends RadixTabsListProps {}
+
+export const TabsList: React.FC<ITabsListProps> = (props) => {
```
[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsList/tabsList.tsx:8]: But then we would need to pass the Tabs.Triggers values via props as an array of objects? And then just wrap Tabs.Content with the root as necessary? I feel like this suggestion may deviate from Radix docs/mental model quite a bit.

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx (hunk 3 above)
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:14]: It's required by Radix by default so I would say yes. I'm only including here to extend documentation as it's an essential implementation detail for the broader component module.

File: src/core/components/tabs/tabsList/tabsList.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Fabricevladimir) [reply src/core/components/tabs/tabsList/tabsList.tsx:8]: Why would we pass it as an array of objects? TabRoot would just take children that it renders like so:
```jsx
<TabRoot>
<TabTrigger value="1" />
<TabContent value="1"/>
</TabRoot>
```

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+export const TabsTrigger: React.FC<ITabsTriggerProps> = (props) => {
+    const { label, iconRight, className, ...otherProps } = props;
+    const { isUnderlined } = useContext(TabsContext);
+
+    const triggerClassNames = classNames(
+        'group flex cursor-pointer items-center gap-x-4 border-primary-400 py-3 text-base font-normal leading-tight text-neutral-500', // base
+        'hover:text-neutral-600', // hover
+        'active:data-[state=active]:text-neutral-800 active:data-[state=active]:shadow-[inset_0_0_0_0,0_1px_0_0] active:data-[state=active]:shadow-primary-400', // active click
+        'focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset', // focus
```
[the remark this answers is not in the document]
Comment (Fabricevladimir) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:29]: Yeah the focus outline

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:10]: Yeah, I wondered should label be optional. Let's keep it required for now

File: src/core/components/tabs/tabsList/tabsList.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsList/tabsList.tsx:8]: Tabs.List only wraps the Triggers and has it's own error boundary for such.

[the remark this answers is not in the document]
Comment (Fabricevladimir) [reply src/core/components/tabs/tabsList/tabsList.tsx:8]: Oh that's dumb lol. Disregard my comments

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:29]: right right! will comment over there

current impl here is not to spec which would just be outline-none

[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx]: lets implement without and get a review with selim after his vacay.

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    iconRight?: IconType;
+}
+
+export const TabsTrigger: React.FC<ITabsTriggerProps> = (props) => {
+    const { label, iconRight, className, ...otherProps } = props;
+    const { isUnderlined } = useContext(TabsContext);
+
+    const triggerClassNames = classNames(
+        'group flex cursor-pointer items-center gap-x-4 border-primary-400 py-3 text-base font-normal leading-tight text-neutral-500', // base
```
[the remark this answers is not in the document]
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx]: clamp or truncate? @Fabricevladimir

File: src/core/components/tabs/tabsTrigger/tabsTrigger.test.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                    <Tabs.Trigger {...completeProps} />
+                </Tabs.List>
+            </Tabs.Root>
+        );
+    };
+
+    it('should render without crashing', () => {
+        render(createTestComponent());
+
+        expect(screen.getByText('Tab 1')).toBeInTheDocument();
```
Comment (Fabricevladimir) [inline src/core/components/tabs/tabsTrigger/tabsTrigger.test.tsx:25]: Check for the presence of the trigger instead by using `getByRole`. I believe the trigger is a button. Relying on the default text isn't ideal.

File: src/core/components/tabs/tabsTrigger/tabsTrigger.test.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    });
+
+    it('should pass the correct value prop', () => {
+        const value = 'complex1';
+        render(createTestComponent({ value }));
+
+        const triggerElement = screen.getByText('Tab 1');
```
Comment (Fabricevladimir) [inline src/core/components/tabs/tabsTrigger/tabsTrigger.test.tsx:32]: See previous comment

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx (hunk 1 above)
Comment (Fabricevladimir) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx:35]: Don't forget this one.

File: src/core/components/tabs/tabsTrigger/tabsTrigger.test.tsx (hunk 8 above)
Comment (thekidnamedkd) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.test.tsx:32]: @Fabricevladimir radix actually injects a 'tab' role 🤝

**Review by Fabricevladimir — APPROVED** [review]

File: src/core/components/tabs/tabsTrigger/tabsTrigger.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Fabricevladimir) [reply src/core/components/tabs/tabsTrigger/tabsTrigger.tsx]: `line-clamp-1`

**Pull request merged**